packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.js

```diff
@@ -58,16 +58,23 @@ function InteractionLayer ({
       return true
     }
 
-    const activeMark = activeTool.createMark({
+    const activeMark = activeTool.onPointerDown({
       id: cuid(),
       frame,
       toolIndex: activeToolIndex
     })
-    activeMark.initialPosition(convertEvent(event))
-    setActiveMark(activeMark)
-    setCreating(true)
-    activeMark.setSubTaskVisibility(false)
-    return false
+
+    if (activeMark.finished) {
+      onPointerMove(event)
+      onFinish(event)
```

Comment:
I think tying `onPointerDown` to finishing a mark will clash with the polygon tool, where you tap multiple times to create a shape. Behaviour for tools and marks should be in the code for those individual types of tools or marks so that it doesn’t interfere with other shapes. 

Having said that, I’ve tried AI4Mars on my phone and it is pretty much impossible to draw a shape with the current polygon tool.